lib/tic_tac_toe_checker/crawler.ex

```diff
@@ -21,6 +22,11 @@ defmodule TicTacToeChecker.Crawler do
   use Finitomata, fsm: @fsm, auto_terminate: true, impl_for: [:on_transition]
 
   @impl Finitomata
+  @spec on_transition(:idle | :ready, :init! | :move!, any(), any) ::
```

Comment:
`@impl Finitomata` already derives specs, being an implementation of `Finitomata` behaviour. It’s ok to make is less generic in some cases, but here is looks a bit redundant.

Reply:
Indeed!